tsconfig.common.json

```diff
@@ -0,0 +1,25 @@
+{
+  "extends": "@tsconfig/node20/tsconfig.json",
+  "exclude": ["node_modules"],
+  "include": ["./jest.config.ts", "./src/**/*.ts", "./__mocks__/**/*.ts"],
+  "compilerOptions": {
+    "rootDir": ".",
+    "sourceMap": true,
+    "strictNullChecks": true,
+    "declaration": false,
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "importHelpers": true,
+    "esModuleInterop": true,
+    "resolveJsonModule": true,
+    "target": "es2022",
+    "module": "nodenext",
+    "moduleResolution": "nodenext",
```

Comment:
If you look at the TS page on recommendations for libraries, they recommend `node16` and not `nodeNext` since `nodeNext` is meant for some future modules implementation (see https://www.typescriptlang.org/docs/handbook/modules/guides/choosing-compiler-options.html#im-writing-a-library).

Also, the default `moduleResolution` will automatically be set to the correct one so it is not needed.